docs/content/development/extensions-core/avro.md

```diff
@@ -66,6 +66,43 @@ This decoder can be used if all the input events can be read using the same sche
 ...
 ```
 
+##### Multiple Inline Schemas Based Avro Bytes Decoder
+
+This decoder can be used if different input events can have different read schema. In that case schema can be specified in the input task json itself as described below.
+
+```
+...
+"avroBytesDecoder": {
+  "type": "schemas_inline",
+  "schemas": {
+    //your id -> schema map goes here, for example
+    "1": {
+      "namespace": "io.druid.data",
+      "name": "User",
+      "type": "record",
+      "fields" [
+        { "name": "FullName", "type": "string" },
+        { "name": "Country", "type": "string" }
+      ]
+    },
+    "2": {
+      "namespace": "io.druid.otherdata",
+      "name": "UserIdentity",
+      "type": "record",
+      "fields" [
+        { "name": "Name", "type": "string" },
+        { "name": "Location", "type": "string" }
+      ]
+    },
+    ...
+    ...
+  }
+}
+...
+```
+
+Note that it is essentially a map of integer schema ID to avro schema object. This parser assumes that first 4 bytes in the input message are integer schema ID and rest is serialized avro message.
```

Comment:
If this is specific to our implementation, we might as well add a version byte too, just in case we want to extend the format later. We may also want to mention what people would have to do in order to get their already-existing avro producers to add the schema id.